src/nexmark/queries/q3.rs

```diff
@@ -0,0 +1,160 @@
+use super::NexmarkStream;
+use crate::{nexmark::model::Event, operator::FilterMap, Circuit, OrdZSet, Stream};
+
+/// Local Item Suggestion
+///
+/// Who is selling in OR, ID or CA in category 10, and for what auction ids?
+/// Illustrates an incremental join (using per-key state and timer) and filter.
+///
+/// See https://github.com/nexmark/nexmark/blob/v0.2.0/nexmark-flink/src/main/resources/queries/q3.sql
+
+const STATES_OF_INTEREST: &[&str] = &["OR", "ID", "CA"];
+const CATEGORY_OF_INTEREST: usize = 10;
+
+pub fn q3(
+    input: NexmarkStream,
+) -> Stream<Circuit<()>, OrdZSet<(String, String, String, u64), isize>> {
```

Comment:
I guess the output could also use static strings in this case.  In general we probably want to use interned strings to represent SQL strings. We rely on interning a lot in DDlog via the [`internment`](https://crates.io/crates/internment) crate. It saves both CPU and memory.